src/client.rs

```diff
@@ -18,41 +18,58 @@
 //! and parsing responses
 //!
 
+use std::{error, io};
 use std::collections::HashMap;
-use std::io;
-use std::io::Read;
 use std::sync::{Arc, Mutex};
 
-use hyper;
-use hyper::client::Client as HyperClient;
-use hyper::header::{Authorization, Basic, ContentType, Headers};
 use serde;
+use base64;
+use http;
 use serde_json;
 
 use super::{Request, Response};
 use util::HashableValue;
 use error::Error;
 
+/// An interface for an HTTP roundtripper that handles HTTP requests.
+pub trait HttpRoundTripper {
+    /// The type of the http::Response body.
+    type ResponseBody: io::Read;
+    /// The type for errors generated by the roundtripper.
+    type Err: error::Error;
+
+    /// Make an HTTP request.  In practice only POST request will be made.
+    fn request(
```

Comment:
Let's call this `post` rather than `request`. If this library is used more generally we may want `get` and `patch` in the future.